src/ray/gcs/store_client/in_memory_store_client.cc

```diff
@@ -89,6 +89,27 @@ Status InMemoryStoreClient::AsyncBatchDelete(const std::string &table_name,
   return Status::OK();
 }
 
+Status InMemoryStoreClient::AsyncGetByIndex(
+    const std::string &table_name, const std::string &index_key,
+    const MapCallback<std::string, std::string> &callback) {
+  auto table = GetOrCreateTable(table_name);
+  absl::MutexLock lock(&(table->mutex_));
+  auto iter = table->index_keys_.find(index_key);
+  std::unordered_map<std::string, std::string> result;
+  if (iter != table->index_keys_.end()) {
+    for (auto &key : iter->second) {
+      auto kv_iter = table->records_.find(key);
+      if (kv_iter != table->records_.end()) {
+        result[kv_iter->first] = kv_iter->second;
+      }
+    }
+    main_io_service_.post([result, callback]() { callback(result); });
+  } else {
+    main_io_service_.post([result, callback]() { callback(result); });
```

Comment:
can remove `else` and just put this line after the if block

Reply:
ok!